packages/app-bridge/src/helpers/link.ts

```diff
@@ -0,0 +1,56 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import type { CoverPage, Document, DocumentCategory, DocumentPage } from '../types';
+
+export const computeCoverPageLink = ({
+    coverPage,
+    language,
+    portalToken,
+}: {
+    coverPage: CoverPage;
+    language: string;
+    portalToken: Nullable<string>;
+}) => {
+    if (portalToken) {
+        return `/d/${portalToken}/${language}`;
+    }
+
+    return coverPage.url;
+};
+
+export const computeDocumentLink = ({
+    document,
+    language,
+    portalToken,
+}: {
+    document: Document;
+    language: string;
+    portalToken: Nullable<string>;
+}) => {
+    if (portalToken) {
+        return `/d/${portalToken}/${language}/${document.slug}`;
+    }
+
+    return `/document/${document.id}/${language}`;
+};
+
+export const computeDocumentPageLink = ({
+    documentPage,
+    language,
+    portalToken,
+    document,
+    documentCategory,
+}: {
+    documentPage: DocumentPage;
+    document: Document;
+    documentCategory?: DocumentCategory;
+    language: string;
+    portalToken: Nullable<string>;
+}) => {
+    const categoryUrlPart = documentCategory ? `${documentCategory.slug}/` : '';
+    if (portalToken) {
+        return `/d/${portalToken}/${language}/${document.slug}#/${categoryUrlPart}${documentPage.slug}`;
+    }
+
+    return `/document/${document.id}/${language}#/${categoryUrlPart}${documentPage.slug}`;
```

Comment:
Please use `getUrlFragment` for the hash and following parts. Same for line 52 above

Reply:
We are in Brand SDK repository, not Web App. Did you change some routing logic recently?

Reply:
Yes. The `categoryUrlPart` should be `-/` if there is no `documentCategory`
From my current perspective we should avoid generating these urls externally.